source/validate.cpp

```diff
@@ -332,14 +270,34 @@ void DebugInstructionPass(ValidationState_t& _,
   }
 }
 
+void ProcessIds(ValidationState_t& _, const spv_parsed_instruction_t& inst) {
```

Comment:
I agree this should go into SsaPass, and I've tried to put it there in two different ways, but I keep getting lots of test failures because of interactions of `SPV_VALIDATE_ID_BIT` and `SPV_VALIDATE_SSA_BIT`.  I'd rather leave it for the next PR.